### PR TITLE
request ticket price

### LINKED ISSUE #115: Request ticket price

### PR DESCRIPTION
**Pull request by radoslavzeman**
- implement basic get ticket price request (it finally works!)
- connect map with purchase screen
- closes #115 

TODO
- format price per locale
- better parkingEnd/duration handling
- bpk, npk, and more

Note:
- TS errors in settings will be solved on BE (optional values will be required again)

### CODE REVIEW

File: app/purchase/index.tsx [hunk 1]
```diff
@@ -45,25 +41,35 @@ const PurchaseScreen = () => {
 
   const chosenVehicle = licencePlate ? getVehicle(licencePlate) : defaultVehicle
 
-  const parkingEnd = addMinutes(new Date(), ticketPriceRequest?.duration ?? 0).toISOString()
+  const parkingEnd = new Date(
+    Date.now() + (ticketPriceRequest?.duration ?? 0) * 60_000,
```
Comment (lukybil) [inline app/purchase/index.tsx:45]: I think we have already discussed this but is it not possible to buy a ticket for future parking?

**Review by lukybil — APPROVED** [review]
😁👍

File: components/map/MapZoneBottomSheet.tsx [hunk 2]
```diff
@@ -77,11 +80,17 @@ const MapZoneBottomSheet = forwardRef<BottomSheet, Props>((props, ref) => {
...
+        ...prev,
+        udr: zone?.UDR_ID.toString(),
+      }))
+
       if (zone === null) {
         localRef.current?.collapse()
       }
     }
-  }, [isFullHeightEnabled, zone])
+  }, [isFullHeightEnabled, setTicketPriceRequest, zone])
```
[the remark this answers is not in the document]
Comment (lukybil) [reply components/map/MapZoneBottomSheet.tsx:93]: I found a way to get rid of the `useEffect`, I can implement it in my next PR which already involves small Map changes, so you can move this to the `handleZoneChange`

File: app/purchase/index.tsx (hunk 1 above)
Comment (radoslavzeman) [reply app/purchase/index.tsx:45]: No, it isn't possible.

File: components/map/MapZoneBottomSheet.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (radoslavzeman) [reply components/map/MapZoneBottomSheet.tsx:93]: kay thanks for feedback, I'll move it :)

**Pull request merged**